packages/context/test/unit/binding.ts

```diff
@@ -162,6 +162,47 @@ describe('Binding', () => {
     });
   });
 
+  describe('withDependencies(Dependencies)', () => {
+    it('binds dependencies without overlap', async () => {
+      const otherKey = 'bar';
+      ctx
+        .bind(key)
+        .toClass(FakeProduct)
+        .withDependencies({
+          config: {
+            foo: 'bar',
+          },
+        });
+
+      ctx
+        .bind(otherKey)
+        .toClass(FakeProduct)
+        .withDependencies({
+          config: {
+            foo: 'baz',
+          },
+        });
+
+      const product1 = (await ctx.get(key)) as FakeProduct;
+      const product2 = (await ctx.get(otherKey)) as FakeProduct;
+
+      function checkConfig(prod: FakeProduct, expected: string) {
+        expect(prod).to.not.be.null();
+        expect(prod.config).to.not.be.null();
+        expect(prod.config.foo).to.equal(expected);
+      }
+
+      // Check in reverse order (in case the 2nd binding is masking the 1st)
+      checkConfig(product2, 'baz');
+      checkConfig(product1, 'bar');
+    });
+
+    class FakeProduct {
```

Comment:
Same goes for class (would be better outside the test to keep it clean unless there's a reason it needs to be inside).